### ISSUE TITLE
Does 2.4.10 really support different index type besides brute-force?

### ISSUE THREAD

**Issue by simshi** [milvus-io/milvus-lite#220]
I looked into the code, segcore_wrapper always create Growing segment, and growing segment only use brute-force in search?
If it's true, do you have plan to support different index type.

**Comment by junjiejiangjjj** [comment]
Yes, since `sealed segment` relies on complex logic such as binlog and compaction, milvus-lite does not integrate, brute-force performs well when the data size is hundreds of thousands, so the support priority for the index is not high.

**Comment by simshi** [comment]
still hope to support various types of indexing, as the index type is an important concept in vector databases (if not a key concept). Perhaps the implementation could be simplified? :flushed:

**Comment by xiaofan-luan** [comment]
We will start work on this once 2.5 is released. 
Please hold this for a while

**Comment by simshi** [comment]
BTW, how about GPU index? is it possible to support?

**Comment by junjiejiangjjj** [comment]
hi @simshi , milvus-lite only supports IVF_FLAT. If you need to use GPU indexing, you need to use Milvus
